### PR TITLE
DYN-4800: Bump Python version to 3.9.12

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

This pull request does:
* Bump pythonnet to `2.5.2`
* Bump embedded python to `3.9.12`

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs [hunk 1]
```diff
@@ -376,6 +376,8 @@ public DSCPythonCodeCompletionProviderCore()
             BasicVariableTypes.Add(Tuple.Create(LIST_VARIABLE, typeof(PyList)));
             BasicVariableTypes.Add(Tuple.Create(DICT_VARIABLE, typeof(PyDict)));
 
+            Python.Included.Installer.SetupPython().Wait();
```
Comment (mjkkirschner) [inline src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs:379]: why is this necessary? The DSCPython engine should do this already, should it not?
I think it's worth testing this change in Revit context if it's really required.

Comment (pinzart90) [reply src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs:379]: There is a note in Dynamo
```
NOTE: Calling SetupPython multiple times will add the install location to the path many times,
        /// potentially causing the environment variable to overflow.
```
I think we might be doing this already ...maybe we should try to only run `SetupPython()` only once ...(keep a isAlreadySetup flag somewhere)
Please search for all occurrences of `SetupPython()`

Comment (sm6srw) [reply src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs:379]: I actually was able to get in here without having python initialized. And we are already calling this method in multiple places. I will take a look.

File: src/Libraries/DSCPython/CPythonEvaluator.cs [hunk 2]
```diff
@@ -336,7 +336,7 @@ import os
         /// NOTE: Calling SetupPython multiple times will add the install location to the path many times,
         /// potentially causing the environment variable to overflow.
         /// </summary>
-        private static void InstallPython()
+        internal static void InstallPython()
```
Comment (sm6srw) [inline src/Libraries/DSCPython/CPythonEvaluator.cs:339]: I made it internal. Would it be OK to make this public?

File: src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs (hunk 1 above)
Comment (sm6srw) [reply src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs]: We are now calling SetupPython() every where. That method maintain a just do it once flag.

**Comment by sm6srw** [comment]
@mjkkirschner PTAL!

File: src/Libraries/DSCPython/CPythonEvaluator.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/Libraries/DSCPython/CPythonEvaluator.cs:339]: since it has a side effect that is potentially dangerous I'd prefer if extensions could not call it.

**Review by mjkkirschner — APPROVED** [review]

**Review by pinzart90 — APPROVED** [review]
